src/charm.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import sys
+sys.path.append('lib')
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, MaintenanceStatus
+from oci_image import OCIImageResource, ResourceError
+
+
+class MultusCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.multus_image = OCIImageResource(self, 'multus-image')
+        self.framework.observe(self.on.start, self.set_pod_spec)
+
+    def set_pod_spec(self, event):
+        if not self.model.unit.is_leader():
+            print('Not a leader, skipping set_pod_spec')
+            self.model.unit.status = ActiveStatus()
+            return
+
+        try:
+            image_details = self.multus_image.fetch()
+        except ResourceError as e:
+            self.model.unit.status = e.status
+            return
```

Comment:
Since you're only observing the `start` (or `install`) event, this will never trigger again on the `upgrade-charm` hook which notifies you of the resource being attached.  You can either observe `self.on.upgrade_charm` or defer the `install` event here instead, in which case it would be re-invoked for every hook until it no longer defers itself.  I recommend the former, since you'll want to properly handle updated resources anyway.

To observe the `upgrade-charm` event, add the following to the `__init__` method:

```python
self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)
```

To defer the `install` event, add this to the except block here, just before the `return`:

```python
event.defer()
```

Note: when you defer an event, it will be re-dispatched on the next hook, as if it had just been triggered.  Deferred events are dispatched prior to new events.  To continue deferring, you must call `event.defer()` each time. Again, though, I'm just putting this here for informational purposes. Deferring is not recommended in this situation, since the `upgrade-charm` hook is the only time the resource availability could actually change.

Reply:
Resolved this by adding `self.framework.observe(self.on.upgrade_charm, self.set_pod_spec)` to `__init__`. Thanks.